### ISSUE TITLE
Support non compliant providers

### ISSUE THREAD

**Issue by sennav** [ramosbugs/oauth2-rs#63]
Hi,

I am writing a CLI for wunderlist.com and using this library to do the oauth authentication. It was everything going well until I realised that wunderlist is not fully compliant with the oauth specs. In their `access_token` response they do not set `token_type` and thus oauth2 fails to transform their response into an object with a parsing error.

What do you think about making `token_type` optional or allowing users to make it optional. I'm a beginner in rust and to solve my issue I simply commented out the `token_type` and I'm using oauth2 that I compile locally but I assume this will happen with other services. I would be happy to help with a PR if you can give me some guidance on how and if you ever want to support this.

**Comment by ramosbugs** [comment]
Hi @sennav,

Unfortunately `token_type` is a mandatory response parameter according to https://tools.ietf.org/html/rfc6749#section-5.1, so I don't think it would be appropriate to make it optional in this library (forcing users to handle `Option` in places that shouldn't be `None`).

That said, the [`TokenResponse`](https://docs.rs/oauth2/2.0.0-beta.2/oauth2/trait.TokenResponse.html) trait allows you to implement your own token response type instead of using the `StandardTokenResponse` struct that is included in the crate. You'll still need the `token_type` method to return something, which could either be a default value you choose or you could try making the `TT` type parameter be an `Option` (not sure if that will work).

Hope that helps!

**Comment by sennav** [comment]
Hi, yes you are right, it would not make sense to force all users to handle an `Option` because some providers are not compliant. I did not know I had the option that you suggested, so thanks for that.

With that said, I'm trying to implement my own token response as you mentioned and I think I got pretty far, but I'm still struggling a bit with rust (as I said, I'm a beginner). Would it be ok to try to submit an example of that then? I guess more beginners like me will try to do something similar.
